dtfs-central-api/src/utils/report-period.ts

```diff
@@ -27,3 +29,40 @@ export const getPreviousReportPeriodStart = ({ month, year }: MonthAndYear): Mon
 
 export const isEqualReportPeriodStart = (reportPeriodStart1: MonthAndYear, reportPeriodStart2: MonthAndYear): boolean =>
   reportPeriodStart1.year === reportPeriodStart2.year && reportPeriodStart1.month === reportPeriodStart2.month;
+
+export const getCurrentReportPeriodForBankSchedule = (bankReportPeriodSchedule: BankReportPeriodSchedule): ReportPeriod => {
+  const previousMonthDate = subMonths(new Date(), 1);
+  const targetMonth = getOneIndexedMonth(previousMonthDate);
+  const targetYear = previousMonthDate.getFullYear();
+
+  const scheduleInCurrentYear = bankReportPeriodSchedule.find((schedule) => targetMonth >= schedule.startMonth && targetMonth <= schedule.endMonth);
+  if (scheduleInCurrentYear) {
+    return {
+      start: {
+        month: scheduleInCurrentYear.startMonth,
+        year: targetYear,
+      },
+      end: {
+        month: scheduleInCurrentYear.endMonth,
+        year: targetYear,
+      },
+    };
+  }
+
+  const scheduleOverlappingYear = bankReportPeriodSchedule.filter((schedule) => schedule.startMonth > schedule.endMonth).at(0);
+  if (!scheduleOverlappingYear) {
+    throw new Error('Failed to get a report period');
+  }
+
+  const reportPeriodIsInPreviousYear = targetMonth <= scheduleOverlappingYear.endMonth;
```

Comment:
While the logic may be correct, I'm not sure that this name is right. In the following scenario where `12` last year is the target
```
   2023 | 2024
11 (12) | 1
```
`reportPeriodIsInPreviousYear` would be false, but the report period _is_ in the previous year. I'm not sure what the correct name would be, and it's a bit hard to follow why this works.

How about this?
```
export const getCurrentReportPeriodForBankSchedule = (bankReportPeriodSchedule: BankReportPeriodSchedule): ReportPeriod => {
  const today = new Date();
  const previousMonthDate = subMonths(today, 1);
  const targetMonth = getOneIndexedMonth(previousMonthDate);
  const targetYear = previousMonthDate.getFullYear();

  const scheduleInCurrentYear = bankReportPeriodSchedule.find((schedule) => targetMonth >= schedule.startMonth && targetMonth <= schedule.endMonth);
  if (scheduleInCurrentYear) {
    return {
      start: {
        month: scheduleInCurrentYear.startMonth,
        year: targetYear,
      },
      end: {
        month: scheduleInCurrentYear.endMonth,
        year: targetYear,
      },
    };
  }

  const scheduleOverlappingYear = bankReportPeriodSchedule.filter((schedule) => schedule.startMonth > schedule.endMonth).at(0);
  if (!scheduleOverlappingYear) {
    throw new Error('Failed to get a report period');
  }

  const currentYear = today.getFullYear();
  const reportPeriodStartsInPreviousYear = targetYear < currentYear || targetMonth <= scheduleOverlappingYear.endMonth;

  return {
    start: {
      month: scheduleOverlappingYear.startMonth,
      year: reportPeriodStartsInPreviousYear ? currentYear - 1 : currentYear,
    },
    end: {
      month: scheduleOverlappingYear.endMonth,
      year: reportPeriodStartsInPreviousYear ? currentYear : currentYear + 1,
    },
  };
};
```

Reply:
Edit: Now I'm not sure if this accounts to the current case, where the entire report period is in the previous year...
```
      2023 | 2024
10 11 (12) |

Reply:
And, further, the report period is actually in the _next_ year relative to the target year

Reply:
I think it becomes a bit clearer if the variable name is `reportPeriodStartsInPreviousYear`

Reply:
Ah, I think the "relative to target year" was the logic I was missing. Yeah, I think if we name this `reportPeriodStartsInPreviousYear` then it makes sense to me 👍